apps/studio/data/database-queues/database-queues-toggle-postgrest-mutation.ts

```diff
@@ -0,0 +1,316 @@
+    queues_public.send(queue_name text, msg jsonb, sleep_seconds integer),
+    queues_public.send(queue_name text, msg jsonb, available_at timestamp with time zone),
+    queues_public.send_batch(queue_name text, msgs jsonb[], sleep_seconds integer),
+    queues_public.send_batch(queue_name text, msgs jsonb[], available_at timestamp with time zone),
+    queues_public.archive(queue_name text, msg_id bigint),
+    queues_public.delete(queue_name text, msg_id bigint),
+    queues_public.read(queue_name text, sleep integer, qty integer)
+  ;
+
+  -- Revoke execute permissions on inner pgmq functions to roles
+  revoke execute on function pgmq.pop(text) from service_role, anon, authenticated;
+  revoke execute on function pgmq.send(text, jsonb, integer) from service_role, anon, authenticated;
+  -- revoke execute on function pgmq.send(payload jsonb, text, text, boolean) from service_role, anon, authenticated;
+  revoke execute on function pgmq.send_batch(text, jsonb[], integer) from service_role, anon, authenticated;
+  -- revoke execute on function pgmq.send_batch(text, jsonb[], timestamp with time zone) from service_role, anon, authenticated;
+  revoke execute on function pgmq.archive(text, bigint) from service_role, anon, authenticated;
+  revoke execute on function pgmq.delete(text, bigint) from service_role, anon, authenticated;
+  revoke execute on function pgmq.read(text, integer, integer) from service_role, anon, authenticated;
+
+  -- For service role, revoke permissions on existing tables
+  revoke all privileges on all tables in schema pgmq from service_role;
+
+  -- Ensure service_role has no permissions on future tables
+  alter default privileges in schema pgmq revoke all privileges on tables from service_role;
+
+  drop schema if exists queues_public;
+`)
+
+export async function toggleQueuesExposurePostgrest({
+  projectRef,
+  connectionString,
+  enable,
+}: DatabaseQueueExposePostgrestVariables) {
+  const sql = enable ? EXPOSE_QUEUES_TO_POSTGREST_SQL : HIDE_QUEUES_FROM_POSTGREST_SQL
+
+  const { result } = await executeSql({
+    projectRef,
+    connectionString,
+    sql,
+    queryKey: ['toggle-queues-exposure'],
+  })
+
+  return result
+}
+
+type DatabaseQueueExposePostgrestData = Awaited<ReturnType<typeof toggleQueuesExposurePostgrest>>
+
+export const useDatabaseQueueToggleExposeMutation = ({
+  onSuccess,
+  onError,
+  ...options
+}: Omit<
+  UseMutationOptions<
+    DatabaseQueueExposePostgrestData,
+    ResponseError,
+    DatabaseQueueExposePostgrestVariables
+  >,
+  'mutationFn'
+> = {}) => {
+  const queryClient = useQueryClient()
+
+  return useMutation<
+    DatabaseQueueExposePostgrestData,
+    ResponseError,
+    DatabaseQueueExposePostgrestVariables
+  >((vars) => toggleQueuesExposurePostgrest(vars), {
+    async onSuccess(data, variables, context) {
+      const { projectRef } = variables
+      await queryClient.invalidateQueries(databaseQueuesKeys.exposePostgrestStatus(projectRef))
```

Comment:
We should also invalidate the schemas query here since we're creating a new schema right?